src/SPIN/fix_neb_spin.cpp

```diff
@@ -0,0 +1,1008 @@
+    } else if (StandardNEB) {
+      prefactor = -dot + kspring*(nlen-plen);
+    }
+
+    if (FinalAndInterWithRespToEIni && veng<vIni) {
+      for (int i = 0; i < nlocal; i++)
+       if (mask[i] & groupbit) {
+          fm[i][0] = 0;
+          fm[i][1] = 0;
+          fm[i][2] = 0;
+        }
+      prefactor =  kspring*(nlen-plen);
+    }
+  }
+
+  for (int i = 0; i < nlocal; i++)
+    if (mask[i] & groupbit) {
+      fm[i][0] += prefactor*tangent[i][0];
+      fm[i][1] += prefactor*tangent[i][1];
+      fm[i][2] += prefactor*tangent[i][2];
+    }
+
+  // project GNEB force on tangent space
+
+  double fdots;
+  for (int i = 0; i < nlocal; i++)
+    if (mask[i] & groupbit) {
+      fdots = fm[i][0]*sp[i][0] + fm[i][1]*sp[i][1] +
+        fm[i][2]*sp[i][2];
+      fm[i][0] -= fdots*sp[i][0];
+      fm[i][1] -= fdots*sp[i][1];
+      fm[i][2] -= fdots*sp[i][2];
+    }
+
+}
+
+/* ----------------------------------------------------------------------
+   geodesic distance calculation (Vincenty's formula)
+------------------------------------------------------------------------- */
+
+double FixNEBSpin::geodesic_distance(double spi[3], double spj[3])
+{
+  double dist;
+  double crossx,crossy,crossz;
+  double dotx,doty,dotz;
+  double normcross,dots;
+
+  crossx = spi[1]*spj[2]-spi[2]*spj[1];
+  crossy = spi[2]*spj[0]-spi[0]*spj[2];
+  crossz = spi[0]*spj[1]-spi[1]*spj[0];
+  normcross = sqrt(crossx*crossx + crossy*crossy + crossz*crossz);
+
+  dotx = spi[0]*spj[0];
+  doty = spi[1]*spj[1];
+  dotz = spi[2]*spj[2];
+  dots = dotx+doty+dotz;
+
+  if (normcross == 0.0 && dots == 0.0)
+    error->all(FLERR,"Incorrect calc. of geodesic_distance in Fix NEB/spin");
+
+  dist = atan2(normcross,dots);
+
+  return dist;
+}
+
+/* ----------------------------------------------------------------------
+   send/recv NEB atoms to/from adjacent replicas
+   received atoms matching my local atoms are stored in xprev,xnext
+   replicas 0 and N-1 send but do not receive any atoms
+------------------------------------------------------------------------- */
+
+void FixNEBSpin::inter_replica_comm()
+{
+  int i,m;
+  MPI_Request request;
+  MPI_Request requests[2];
+  MPI_Status statuses[2];
+
+  // reallocate memory if necessary
+
+  if (atom->nmax > maxlocal) reallocate();
+
+  double **x = atom->x;
+  double **f = atom->f;
+  double **sp = atom->sp;
+  double **fm = atom->fm;
+  tagint *tag = atom->tag;
+  int *mask = atom->mask;
+  int nlocal = atom->nlocal;
+
+  // -----------------------------------------------------
+  // 3 cases: two for single proc per replica
+  //          one for multiple procs per replica
+  // -----------------------------------------------------
+
+
+  // case 1 => to be done
+
+  // single proc per replica
+  // all atoms are NEB atoms and no atom sorting
+  // direct comm of x -> xprev and x -> xnext
+
+  if (cmode == SINGLE_PROC_DIRECT) {
+    if (ireplica > 0)
```

Comment:
These if statements look suspicious and the indentation does not match the logic of then code. if only one MPI call should be covered by the ifs, then the second MPI call should be properly indented (i.e. 2 spaces less). if both MPI calls should be covered by the ifs, then there must be curly braces added.